From 71a05adbfaea0c5931a091a777f3e16c086064fc Mon Sep 17 00:00:00 2001
From: zer0921 <zer0921@126.com>
Date: Sat, 31 May 2025 19:15:49 +0800
Subject: [PATCH 2/8] adapt python 3.10

---
 neutron/__init__.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/neutron/__init__.py b/neutron/__init__.py
index 7e68d59..a7b76e8 100644
--- a/neutron/__init__.py
+++ b/neutron/__init__.py
@@ -29,3 +29,8 @@ else:
 six.moves.builtins.__dict__['_'] = removals.remove(
     message='Builtin _ translation function is deprecated in OpenStack; '
             'use the function from _i18n module for your project.')(_)  # noqa
+
+# python lang monkey_patch
+import collections
+collections.Sequence = collections.abc.Sequence
+collections.Mapping = collections.abc.Mapping
-- 
2.34.1

