From 02d9292d3eafd658ee94264c7065eb156f53bab7 Mon Sep 17 00:00:00 2001
From: zer0921 <zer0921@126.com>
Date: Sat, 31 May 2025 19:24:51 +0800
Subject: [PATCH 5/8] server support 0 rpc_worker

---
 neutron/service.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/neutron/service.py b/neutron/service.py
index 0a5fafe..48cbb47 100644
--- a/neutron/service.py
+++ b/neutron/service.py
@@ -177,8 +177,6 @@ def _get_rpc_workers(plugin=None):
     if workers is None:
         # By default, half as many rpc workers as api workers
         workers = int(_get_api_workers() / 2)
-    if workers < 1:
-        workers = 1
 
     # If workers > 0 then start_rpc_listeners would be called in a
     # subprocess and we cannot simply catch the NotImplementedError.  It is
-- 
2.34.1

